### PR TITLE
IPlyska - Task2

### PR DESCRIPTION
**Pull request by StrangerM**
Task2 -done

### CODE REVIEW

File: Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+            else
+            {
+                Console.WriteLine($"Result of parsing: true, Point =({point.Value.X},{point.Value.Y})");
+            }
+        }
+
+        /// <summary>
+        /// змінна index - це змінна класу, а color - обєкту.
+        /// поведінка статичних змінних подібна до поведінки сінглтона.
+        /// Відповідно коли ми ініціалізуємо статичну змінну виділяється ячейка в
```
Comment (olehkrut) [inline Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs:34]: Nice, overall it's right, static fields are like singleton,
deep dive: but all static fields are actually stored in the heap, to be more precise `Type` object(the one which returned with `GetType()`) references them

File: Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        /// </summary>
+        /// <param name="color"></param>
+        /// <returns></returns>
+        static Marker? InitMarker(string color)
+        {
+            if (color is null)
+            {
+                return null;
+            }
+            return new Marker { color = color };
```
Comment (olehkrut) [inline Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs:58]: new line before return pls)

File: Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        {
+            int[] testArray = { 2, 5, 7 };
+
+            Console.WriteLine("The values store into the array are");
+            foreach (var item in testArray)
+            {
+                Console.Write(item);
+            }
+
+            testArray = testArray.Reverse().ToArray();
```
Comment (olehkrut) [inline Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs:89]: he-he, LINQ is good and all, but the idea was to use `Array.Reverse(testArray)` - this one, won't allocate a new array, but modify the existing one, here is an implementation if you are interested - https://referencesource.microsoft.com/#mscorlib/system/array.cs,f5edebc3412b666b

File: Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+        static void CountDuplicate(int[] testArray)
+        {
+            if (testArray is null || testArray.Count() == 0)
+            {
+                Console.WriteLine("Array can not be null or an empty");
+                return;
+            }
+
+            var secondArray = testArray.Distinct().ToArray();
+            Console.WriteLine("Total number of duplicate elements found in the array is " + (testArray.Length - secondArray.Length));
```
Comment (olehkrut) [inline Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs:112]: nice)

File: Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+        }
+
+        #endregion
+
+        /// <summary>
+        /// swap the values of two integer numbers
+        /// if need can use ref parameter
```
Comment (olehkrut) [inline Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs:119]: yeah, the idea was to use `ref` but it's fine, I see that you understand)

File: Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+        public Marker(string Color)
+        {
+            color = Color;   
+        }
+        public void PrintMarker()
+        {
+            Console.WriteLine(this.color + index);
+        }
+        public void PrintMarker(Marker marker)
```
Comment (olehkrut) [inline Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs:151]: better to make this method static in this case

**Review by olehkrut — COMMENTED** [review]
Really nice, well done!

File: Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs (hunk 2 above)
Comment (StrangerM) [reply Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs:59]: added)

File: Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs (hunk 3 above)
Comment (StrangerM) [reply Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs]: changed to static method of Array class. Thanks for link

File: Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs (hunk 6 above)
Comment (StrangerM) [reply Bootcamp/Lecture2/Tasks/IuriiPlLect2/Program.cs]: also changed method to static

**Review by olehkrut — APPROVED** [review]

**Review by m96roman — COMMENTED** [review]
Well done

**Pull request merged**